File: CK3toEU4/Source/CK3World/Dynasties/HouseNameScraping.cpp
#include "HouseNameScraping.h"
#include "CommonRegexes.h"
#include "Log.h"
#include "ParserHelpers.h"

CK3::HouseNameScraping::HouseNameScraping(std::istream& theStream)
{
	registerKeys();
	parseStream(theStream);
	clearRegisteredKeywords();
}

void CK3::HouseNameScraping::registerKeys()
{
	registerKeyword("name", [this](std::istream& theStream) {
		name = commonItems::getString(theStream);
	});
	registerKeyword("prefix", [this](std::istream& theStream) {
		prefix = commonItems::getString(theStream);
	});
	registerRegex(commonItems::catchallRegex, commonItems::ignoreItem);
}

File: CK3ToEU4/Source/CK3/Dynasties/HouseNameScraping.cs

namespace CK3
{
class HouseNameScraping
{
	public HouseNameScraping() {}
	public HouseNameScraping(BufferedReader reader)
	{
		registerKeys();
		parseStream(theStream);
		clearRegisteredKeywords();
	}
	public const auto& getPrefix() const { return prefix; }
	public const auto& getName() const { return name; }

  private:
	void registerKeys()
	{
		registerKeyword("name", [this](BufferedReader reader) {
			name = reader.GetString();
		});
		registerKeyword("prefix", [this](BufferedReader reader) {
			prefix = reader.GetString();
		});
		registerRegex(CommonRegexes.Catchall, ParserHelpers.IgnoreItem);
	}

	string name;
	string prefix;
};
}